commit 1dab13d3d5805b795690a693a74caef5bb5e6caa (refs/changes/51/4951/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-10-12T11:40:18-07:00 (12 months ago)
    
    TRITON-867 add 'make cutarelease' convenience to node-vmadm for publishing and tagging

diff --git a/Makefile b/Makefile
index 523c6bf..ced25ca 100644
--- a/Makefile
+++ b/Makefile
@@ -65,6 +65,25 @@ test-coverage-unit: | $(TAP) $(STAMP_NODE_MODULES) $(BUILD)
 	$(TAP) --jobs=$(TEST_UNIT_JOBS) --output-file=$(BUILD)/test.unit.tap --coverage \
 		test/unit/**/*.test.js
 
+.PHONY: cutarelease
+cutarelease:
+	[[ -z `git status --short` ]]  # If this fails, the working dir is dirty.
+	@which json 2>/dev/null 1>/dev/null && \
+	    ver=$(shell json -f package.json version) && \
+	    name=$(shell json -f package.json name) && \
+	    publishedVer=$(shell npm view -loglevel silent -j $(shell json -f package.json name)@$(shell json -f package.json version) version 2>/dev/null) && \
+	    if [[ -n "$$publishedVer" ]]; then \
+		echo "error: $$name@$$ver is already published to npm"; \
+		exit 1; \
+	    fi && \
+	    echo "** Are you sure you want to tag and publish $$name@$$ver to npm?" && \
+	    echo "** Enter to continue, Ctrl+C to abort." && \
+	    read
+	ver=$(shell cat package.json | json version) && \
+	    date=$(shell date -u "+%Y-%m-%d") && \
+	    git tag -a "v$$ver" -m "version $$ver ($$date)" && \
+	    git push --tags origin && \
+	    npm publish
 
 include ./tools/mk/Makefile.deps
 include ./tools/mk/Makefile.targ
